### ISSUE TITLE
Failed to recognize change in props.

### ISSUE THREAD

**Issue by joncppl** [brownieboy/react-bootstrap-slider#19]
My application requires that the range for the slider change based on the state of the application.

Specifically I'm changing the integer value of the prop 'max' between 10 and 25 depending on my application's state.

``` javascript
                <Slider
                  value={scale}
                  max={cut ? 10 : 25}
                  min={2}
                  step={1}
                  slideStop={e => updateScale(e.target.value)}
                />
```

However, the slider doesn't update for this change.

I quickly fiddled with the code in the library and found that removing the if statement around

``` javascript
                if (this.mySlider.max) {
                    this.mySlider.setAttribute("max", this.props.max);
                }
```

fixes the issue and it updates as I would expect. (It seems that this.mySlider doesn't have the property max)

Am I doing something wrong, or does this library need to updated?

**Comment by brownieboy** [comment]
I'm not sure why it's not working for you.  If you check the demo code in that changing the max property on the fly is one of the things I do as part of the Change Axes buttons. 

Is your application on Github or somewhere else that I could see more of it?

**Comment by joncppl** [comment]
I've run the demo you mentioned after a fresh npm install and I see the same issue. clicking change axes updates the value but does not change the axes min/max like it should. 

peeking into the mySlider object, there is no property max (or min), so the if statement I mentioned before never gets triggered because this.mySlider.max is always undefined. The max/min value is correctly in mySlider.options though. 

![img](http://i.imgur.com/6Qn5lbw.png)

To get it to work for me I may just fork the repo and change those if statements to this.mySlider.options.max

FWIW, I'm using it [here](https://bitbucket.org/joncppl/acc-e/src/123ebea6474479010e1e277cde31dfee6e96c067/app/components/AnimeListSettings.js?at=master&fileviewer=file-view-default#AnimeListSettings.js-185)

**Comment by brownieboy** [comment]
Ok, but I probably won't have time to look at it again until the weekend.

I suggest that you fork for now.  If you then want to submit a Pull Request, I'll merge your changes back into the package.

**Comment by brownieboy** [comment]
Ok, I finally found some time to look at this in more detail.

You were right:  the `if (this.mySlider.max) {` test was doing nothing, so should be removed as you have done in your pull request #20.  This messed up the Ticks demo/test though, because the code was then trying to set max, min and step attributes even when they weren't being passed in as props, as was the case with the Ticks demo.  (You don't need those attributes when using Ticks.)

The solution, I think, it to test for the presence of those props before trying to set attributes based on them, like so:

```JavaScript
        if (this.props.min) {
            this.mySlider.setAttribute("min", this.props.min);
        }
        if (this.props.max) {
            this.mySlider.setAttribute("max", this.props.max);
        }
        if (this.props.step) {
            this.mySlider.setAttribute("step", this.props.step);
        }
```

I'll get this rolled into an update soon.

**Comment by brownieboy** [comment]
Just published version 1.1.3 of the package to npmjs, which fixes this issue.  The final version of the relevant code incorporates your changes, like so:

```JavaScript
        if (this.props.min && (this.mySlider.min || this.mySlider.options.min)) {
            this.mySlider.setAttribute("min", this.props.min);
        }
        if (this.props.max && (this.mySlider.max || this.mySlider.options.max)) {
            this.mySlider.setAttribute("max", this.props.max);
        }
        if (this.props.step && (this.mySlider.step || this.mySlider.options.step)) {
            this.mySlider.setAttribute("step", this.props.step);
        }
```

Many thanks for your help.
